notebook/nbextensions.py

```diff
@@ -208,24 +185,89 @@ def install_nbextension(path, overwrite=False, symlink=False, user=False, prefix
                 dest_dir = pjoin(full_dest, parent[len(path):])
                 if not os.path.exists(dest_dir):
                     if verbose >= 2:
-                        print("making directory %s" % dest_dir)
+                        log("Making directory: %s" % dest_dir)
                     os.makedirs(dest_dir)
                 for file in files:
                     src = pjoin(parent, file)
-                    # print("%r, %r" % (dest_dir, file))
+                    # self.log.info("%r, %r" % (dest_dir, file))
                     dest_file = pjoin(dest_dir, file)
-                    _maybe_copy(src, dest_file, verbose)
+                    _maybe_copy(src, dest_file, verbose, log=log)
         else:
             src = path
-            _maybe_copy(src, full_dest, verbose)
+            _maybe_copy(src, full_dest, verbose, log=log)
+
+
+def install_nbextension_python(package, overwrite=False, symlink=False,
+                        user=False, sys_prefix=False, prefix=None, nbextensions_dir=None,
+                        verbose=1, log=None):
+    """Install an nbextension bundled in a Python package."""
+    if log is None: log = print
+    m, nbexts = _get_nbextension_metadata(package)
+    base_path = os.path.split(m.__file__)[0]
+    for nbext in nbexts:
+        src = os.path.join(base_path, nbext['src'])
+        dest = nbext['dest']
+        require = nbext['require']
+        log(src, dest, require)
+        install_nbextension(src, overwrite=overwrite, symlink=symlink,
+            user=user, sys_prefix=sys_prefix, prefix=prefix, nbextensions_dir=nbextensions_dir,
+            destination=dest, verbose=verbose, log=log
+            )
+
+
+def enable_nbextension_python(package, user=False, sys_prefix=False, log=None):
+    """Enable an nbextension associated with a Python package."""
+    if log is None: log = print
+    data = _read_config_data(user=user, sys_prefix=sys_prefix)
+    module, nbexts = _get_nbextension_metadata(package)
+    for nbext in nbexts:
+        require = nbext['require']
+        section = nbext['section']
```

Comment:
what about making section a list in the metadata??

Suppose the user has one extension that should be "activated" in the notebook and the tree section... if we do not provide this, the user has to enable it in another subsequent step with the `jupyter nbextension enable blabla`... again... I feel that we should handle this case...